addons/cookoff/XEH_postInit.sqf

```diff
@@ -108,3 +108,59 @@ if (isServer) then {
         } forEach allCurators;
     }] call CBA_fnc_addEventHandler;
 };
+
+if (hasInterface) then {
+    GVAR(soundObjPool) = [];
+
+    [QFUNC(poolGetObject), {
+        private _pooledObject = GVAR(soundObjPool) param [0, [-999, objNull]];
+        _pooledObject params ["_time", "_obj"];
+        if (isNull _obj || _time >= time) then {
+            _obj = "#particlesource" createVehicleLocal [0, 0, 0];
+        } else {
+            GVAR(soundObjPool) deleteAt 0;
+        };
+        GVAR(soundObjPool) pushBack [time + 8, _obj];
+        _obj; // return
+    }] call CBA_fnc_compileFinal;
+
+    [QGVAR(playCookoffSound), {
+        params ["_obj", "_sound", "_maxDis"];
+        if (isNull _obj) exitWith {};
+        private _distance = _obj distance (positionCameraToWorld [0,0,0]);
+        if (_distance > _maxDis) exitWith {};
+        _distance = _distance + (random (50) - 25);
+        private _dis = switch (true) do {
```

Comment:
``` 
private _dis = if (_distance < 325) then {
    "close"
} else {
    if (_distance < 952) then {
        "mid"
    } else {
        "far"
    }
};
```
Less code and 33% faster.

Reply:
This might be even faster:
```sqf
private _dis = [["far", "mid"] select (_distance < 952), "close"] select (_distance < 325);
```
Can do multi-line for readability.

BTW, you can use `sqf` as highlighter, it works on GitHub.

Reply:
`["close", "mid", "far"] select linearConversion [11.5, 1265.5, _distance, 0, 2, true]`
:smile:

Reply:
Jonas wins. His code is almost twice as fast as mine and a breath faster than PabstMirror's.

Reply:
Ok i am currently not able to change it feel free to change